timeseries/src/autogluon/timeseries/models/statsmodels/models.py

```diff
@@ -220,3 +221,147 @@ def _predict_with_local_model(
+    n_jobs : int or float, default = 0.5
+        Number of CPU cores used to fit the models in parallel.
+        When set to a float between 0.0 and 1.0, that fraction of available CPU cores is used.
+        When set to a positive integer, that many cores are used.
+        When set to -1, all CPU cores are used.
+    """
+
+    statsmodels_allowed_init_args = [
+        "deseasonalize",
+        "seasonal_period",
+        "use_test",
+        "method",
+        "difference",
+    ]
+    statsmodels_allowed_fit_args = []
+
+    def _update_sm_model_init_args(self, sm_model_init_args: dict, data: TimeSeriesDataFrame) -> dict:
+        sm_model_init_args = sm_model_init_args.copy()
+        sm_model_init_args.setdefault("deseasonalize", True)
+
+        # Infer seasonal_period if seasonal_period is not given / is set to None
+        seasonal_period = sm_model_init_args.pop("seasonal_period", None)
+        if seasonal_period is None:
+            seasonal_period = get_seasonality(data.freq)
+        sm_model_init_args["period"] = seasonal_period
+
+        return sm_model_init_args
+
+    def _fit_local_model(
+        self, timeseries: pd.Series, sm_model_init_args: dict, sm_model_fit_args: dict
+    ) -> FittedLocalModel:
+        # ThetaModel in statsmodels doesn't provide a way to initialize the model with trained parameters,
+        # so we delegate training to `_predict_with_local_model`
+        if sm_model_init_args["deseasonalize"] and len(timeseries) < 2 * sm_model_init_args["period"]:
+            sm_model_init_args = sm_model_init_args.copy()
+            sm_model_init_args["deseasonalize"] = False
+        return FittedLocalModel(model_name=self.name, sm_model_init_args=sm_model_init_args, parameters=None)
+
+    def _predict_with_local_model(
+        self, timeseries: pd.Series, fitted_model: FittedLocalModel, quantile_levels: List[float]
+    ) -> pd.DataFrame:
+        assert fitted_model.model_name == self.name
+        timeseries.index.freq = self.freq
+        with statsmodels_warning_filter():
+            base_model = StatsmodelsTheta(endog=timeseries, **fitted_model.sm_model_init_args)
+            sm_model = base_model.fit(disp=False)
+        # The API is inconsistent with ETS/ARIMA, so we access the predictions differently here
+        results = [sm_model.forecast(self.prediction_length).rename("mean")]
+        for q in quantile_levels:
+            if q < 0.5:
+                coverage = 2 * q
+                column_index = 0
+            else:
+                coverage = 2 * (1 - q)
+                column_index = 1
+            quantile_pred = sm_model.prediction_intervals(steps=self.prediction_length, alpha=coverage)
+            # Select lower bound of the confidence interval if q < 0.5, upper bound otherwise
+            results.append(quantile_pred.iloc[:, column_index].rename(str(q)))
+        return pd.concat(results, axis=1)
+
+
+class SeasonalNaiveModel(AbstractStatsmodelsModel):
```

Comment:
I have a concern about putting this under the statsmodels namespace just to be able to use the `AbstractStatsmodelsModel` type. I think we can redesign now if necessary to have `AbstractLocalModel` `AbstractStatsmodelsModel < AbstractLocalModel` and not mix this one in the Statsmodels namespace. This is both semantically confusing (i.e., what if the test for this model fails due to a dependency quirk in the statsmodels import statements?) and also makes dependency management quite difficult (e.g., what if we want to lazily import statsmodels, will we also not import SeasonalNaive?).

Reply:
Makes sense, I also had some other ideas about optimizing local models that we can discuss in the other PR (though, probably not highest priority right now)